### PR TITLE
Unit test for Magento\Catalog\Observer\SetSpecialPriceStartDate has added

### PR DESCRIPTION
**Pull request by mmezhensky**
### Description (*)

This PR adds unit test that covers Magento\Catalog\Observer\SetSpecialPriceStartDate

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+namespace Magento\Catalog\Test\Unit\Observer;
+
+use Magento\Framework\Event;
+use Magento\Framework\Event\Observer;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use Magento\Framework\Stdlib\DateTime\Timezone;
+use Magento\Catalog\Model\Product;
+use Magento\Catalog\Observer\SetSpecialPriceStartDate;
+use PHPUnit\Framework\TestCase;
+use PHPUnit\Framework\MockObject\MockObject;
```
Comment (dmytro-ch) [inline app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php:17]: Could you please optimize imports by sorting them alphabetically?

File: app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+            [
+                'localeDate' => $this->timezone
+            ]
+        );
+    }
+
+    /**
+     * Test observer execute method
+     */
+    public function testExecuteModifySpecialFromDate()
```
Comment (dmytro-ch) [inline app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php]:
```suggestion
    public function testExecuteModifySpecialFromDate(): void
```

File: app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+    /**
+     * @var \DateTime|MockObject
+     */
+    private $dateObject;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp() : void
```
Comment (dmytro-ch) [inline app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php]:
```suggestion
    protected function setUp(): void
```

File: app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+            ->willReturn($this->productMock);
+
+        $this->dateObject->expects($this->any())
+            ->method('setTime')
+            ->will($this->returnSelf());
+
+        $this->timezone
+            ->expects($this->once())
+            ->method('date')
+            ->will($this->returnValue($this->dateObject));
```
Comment (dmytro-ch) [inline app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php]: I would recommend using `willReturn` for better consistency with other stubs.

```suggestion
            ->willReturn($this->dateObject));
```

File: app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+            ->willReturn($this->eventMock);
+
+        $this->eventMock
+            ->expects($this->once())
+            ->method('getProduct')
+            ->willReturn($this->productMock);
+
+        $this->dateObject->expects($this->any())
+            ->method('setTime')
+            ->will($this->returnSelf());
```
Comment (dmytro-ch) [inline app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php]:
```suggestion
            ->willReturnSelf();
```

**Review by dmytro-ch — DISMISSED** [review]
Hello @mmezhensky , thank you for your contribution.
Could you please fix minor code review recommendations?
Thank you!

File: app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php (hunk 1 above)
Comment (mmezhensky) [reply app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php:17]: @dmytro-ch fixed, please check. Tnx!

File: app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php (hunk 2 above)
Comment (mmezhensky) [reply app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php]: @dmytro-ch fixed, please check. Tnx!

File: app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php (hunk 3 above)
Comment (mmezhensky) [reply app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php]: @dmytro-ch fixed, please check. Tnx!

File: app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php (hunk 4 above)
Comment (mmezhensky) [reply app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php]: @dmytro-ch fixed, please check. Tnx!

File: app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php (hunk 5 above)
Comment (mmezhensky) [reply app/code/Magento/Catalog/Test/Unit/Observer/SetSpecialPriceStartDateTest.php]: @dmytro-ch fixed, please check. Tnx!

**Review by dmytro-ch — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @dmytro-ch, thank you for the review.
[ENGCOM-6984](https://jira.corp.magento.com/browse/ENGCOM-6984) has been created to process this Pull Request
:eight_spoked_asterisk: @dmytro-ch, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Delta** [comment]
**Notice:** QA not applicable

**Pull request merged**